Branch-SDK/src/main/java/io/branch/referral/Branch.java

```diff
@@ -2153,15 +2153,15 @@ public void userCompletedAction(final String action, BranchViewHandler.
      */
     public void userCompletedAction(@NonNull final String action, JSONObject
             metadata, BranchViewHandler.IBranchViewEvents callback) {
-        ServerRequest req = new ServerRequestActionCompleted(context_, action, metadata, callback);
+        ServerRequest req = new ServerRequestActionCompleted(context_, action, null, metadata, callback);
         if (!req.constructError_ && !req.handleErrors(context_)) {
             handleNewRequest(req);
         }
     }
     
     public void sendCommerceEvent(@NonNull CommerceEvent commerceEvent, JSONObject
             metadata, BranchViewHandler.IBranchViewEvents callback) {
-        ServerRequest req = new ServerRequestRActionCompleted(context_, commerceEvent, metadata, callback);
+        ServerRequest req = new ServerRequestActionCompleted(context_, "purchase", commerceEvent, metadata, callback);
```

Comment:
Why isn't this using BRANCH_STANDARD_EVENT::PURCHASE?

In fact, why is there a CurrencyEvent at all?  Everything here except the List<Product> is 1-1 mapped in the base Event class.

Reply:
You probably meant `CommerceEvent` not `CurrencyEvent` and I agree it would be simpler to have a single event class but it's not part of this PR. A separate refactoring PR would be probably be the best way to do it. (I did include some refactoring here but tried to keep it at a minimum so it doesn't interfere with the core thing of this PR, closing the field gaps)